### ISSUE TITLE
iOS Keyboard Misbehavior with TextFields

### ISSUE THREAD

**Issue by buyblvd-ryan** [FlutterFlow/flutterflow-issues#1082]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Currently, several components of TextField behavior aren't working on iOS web (or PWA) with onscreen keyboard.  

1. It is not possible to advance through multiple TextFields on a page, regardless of whether or not the fields are wrapped inside a form (whereas pressing tab on desktop works to advance through the fields).
2. If the TextField widget OnSubmit actions do not trigger on pressing "Done" as should be happening per [documentation](https://docs.flutterflow.io/widgets-and-components/widgets-ui-elements/form-elements-1/textfield).  I am not able to reproduce this issue, in a new project, so it may not be appropriate here.

### Expected Behavior

Pressing "Done" on the iOS keyboard should trigger the OnSubmit action when running in Safari or as PWA.

It should be possible to advance through multiple text fields on a page either by clicking the next/previous arrows on the keyboard (which are currently greyed out), by pressing "Next" or "Done" trigger actions on the keyboard, or upon autofilling a field from the iOS autofill suggestions.

### Steps to Reproduce

1. Create a new page.
2. Add two text fields.
3. Add an onSubmit action to the second text field.
4. Publish to web.
5. Access the page on iOS Safari, cannot auto advance through the fields but for some reason onSubmit does work here but not in my project.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4sls/q4ot1m/UC7r6JK+5r/T0ROVw1U+Y4scpFcxQcB4jZP4kQfPfQP0xoZObnY1FYOFmmp14z0anKuezPJfQ5CzCeUaJJyYd1FwuWeDqQV7WTF7WST2t8RO9bIEig5J6niCRQCMteW1Vg7F2QBunJSgHDQo+yIkQ8Sq/LZO4=

### Context

This one seems pretty self explanatory, but...

1. I would like my create account page with multiple text fields to support auto-advancing to the next text field without having to click into it.
2. My search bar doesn't work on mobile because I cannot trigger "OnSubmit."

### Visual documentation

Screen recordings of both issues shown [here](https://drive.google.com/drive/folders/1guZNAYH7-hBxva0-yNisZnlcvFtV9uzi?usp=drive_link).

### Additional Info

I was referred here for both issues by FlutterFlow chat support.

### Environment

```markdown
- FlutterFlow version: v3.1
- Platform: Web
- Browser name and version: Safari
- Operating system and version affected: iOS 16 and iOS 17 Beta
```

**Comment by SameerShahzad99** [comment]
It looks like the project that you have shared is not a blank project with setup issues, Can you please share a blank project with setup?

**Comment by buyblvd-ryan** [comment]
@SameerShahzad99 Do you mean like this?

https://app.flutterflow.io/project/keyboard-test-vyg1du

**Comment by sbis04** [comment]
I tested the `onSubmit` action on web, Android & iOS (also as a PWA in both), and it seems to function correctly.

And about your question about advancing through multiple TextFields, it's actually not a bug. So, in Flutter by default the TextField action is set to `done` which means it would dismiss the keyboard and remove the current focus (but not traverse through the fields even if it's wrapped inside a Form). And, on web desktop the keyboard tab works in a very different way (so it kinda works).

But hopefully, we will have a property to set the action type of a TextField soon!

Closing this issue as it doesn't look like it's a bug.
